lib/jsdom/living/nodes/Document-impl.js

```diff
@@ -227,6 +227,8 @@ class DocumentImpl extends NodeImpl {
     }
 
     this._contentType = privateData.options.contentType;
+    this._encoding = privateData.options.encoding;
```

Comment:
Very interesting. I like the idea of passing a string => UTF-8, otherwise windows-1252.

I will have to think carefully about this when we redesign the public jsdom API to see if there is a better solution. E.g. if we start accepting binary data instead of just strings, or similar.